packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -0,0 +1,135 @@
+import { Platform } from '@expo/eas-build-job';
+import { ResourceClass } from '@expo/eas-json';
+import chalk from 'chalk';
+import semver from 'semver';
+
+import { DynamicConfigContextFn } from '../../commandUtils/context/DynamicProjectConfigContextField';
+import { BuildResourceClass } from '../../graphql/generated';
+import Log from '../../log';
+import { ProfileData } from '../../utils/profiles';
+import { getReactNativeVersionAsync } from '../metadata';
+
+const iosResourceClassToBuildResourceClassMapping: Record<ResourceClass, BuildResourceClass> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.IosDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.IosLarge,
+  [ResourceClass.M1_EXPERIMENTAL]: BuildResourceClass.IosM1Large,
+  [ResourceClass.M1_MEDIUM]: BuildResourceClass.IosM1Medium,
+  [ResourceClass.M1_LARGE]: BuildResourceClass.IosM1Large,
+  [ResourceClass.INTEL_MEDIUM]: BuildResourceClass.IosIntelMedium,
+  [ResourceClass.MEDIUM]: BuildResourceClass.IosMedium,
+};
+
+const androidResourceClassToBuildResourceClassMapping: Record<
+  Exclude<
+    ResourceClass,
+    | ResourceClass.M1_EXPERIMENTAL
+    | ResourceClass.M1_MEDIUM
+    | ResourceClass.M1_LARGE
+    | ResourceClass.INTEL_MEDIUM
+  >,
+  BuildResourceClass
+> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+  [ResourceClass.MEDIUM]: BuildResourceClass.AndroidMedium,
+};
+
+async function resolveIosDefaultRequestedResourceClassAsync(
```

Comment:
The same with this function. I'd place it below `resolveIosResourceClassAsync` as it's less generic.